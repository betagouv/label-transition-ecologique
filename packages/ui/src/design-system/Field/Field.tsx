import classNames from 'classnames';
import {Icon} from '@design-system/Icon';

export type FieldState =
  | 'default'
  | 'info'
  | 'error'
  | 'success'
  | 'warning'
  | 'disabled';

export const stateToTextColor: Record<FieldState, string> = {
  info: 'text-info-1',
  error: 'text-error-1',
  success: 'text-success-1',
  warning: 'text-warning-1',
  default: 'text-grey-8',
  disabled: 'text-grey-6',
};

const stateToIcon: Record<FieldState, string> = {
  info: 'information-fill',
  error: 'spam-fill',
  success: 'checkbox-circle-fill',
  warning: 'alert-fill',
  default: '',
  disabled: '',
};

type Props = {
  /** Titre */
  title: string;
  /** Enfant: input, select... */
  children: React.ReactNode;
  /** Pour surcharger les styles du container */
  className?: string;
  /** Complément d'informations */
  hint?: string;
  /** Pour lier le libellé et le champ qu'il contient */
  htmlFor?: string;
  /** Réduit la taille du titre */
  small?: boolean;
} & FieldMessageProps;

/** Wrapper pour élément de formulaire donnant des informations et un état */
export const Field = ({
  title,
  className,
  hint,
  htmlFor,
  state = 'default',
  message,
  children,
  small,
}: Props) => {
  return (
    <div className={classNames(`flex-grow flex flex-col gap-3`, className)}>
      <label htmlFor={htmlFor}>
        {/** Title */}
        <div
          className={classNames('font-medium text-grey-8', {
            'text-grey-5': state === 'disabled',
            'text-sm': small,
          })}
        >
          {title}
        </div>
        {/** Hint */}
        {hint !== undefined && (
          <span
            className={classNames('block mt-2 text-xs text-grey-7', {
              'text-grey-5': state === 'disabled',
            })}
          >
            {hint}
          </span>
        )}
      </label>
      {children}
      <FieldMessage state={state} message={message} />
    </div>
  );
};

export type FieldMessageProps = {
  /** État */
  state?: FieldState;
  /** Message d'état affiché en dessous du champ */
  message?: string;
  /** Pour surcharger les styles du container du message */
  messageClassName?: string;
};

/**
 * Affiche un message additionnel au-dessous d'un champ
 */
export const FieldMessage = ({
  message,
  state = 'default',
  messageClassName,
}: FieldMessageProps) => {
  return (
    message !== undefined && (
      <div
        className={classNames(
          'flex items-center',
          stateToTextColor[state],
          messageClassName
        )}
      >
        {state !== 'disabled' && state !== 'default' && (
          <Icon icon={stateToIcon[state]} size="sm" className="mr-1" />
        )}
        <span className="text-xs">{message}</span>
      </div>
    )
  );
};
