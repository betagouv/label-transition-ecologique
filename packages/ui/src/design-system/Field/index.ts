export * from './Field';
export * from './FieldMessage';
