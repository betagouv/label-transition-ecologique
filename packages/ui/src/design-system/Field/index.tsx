import classNames from 'classnames';
import {Icon} from '@design-system/Icon';

export type FieldState =
  | 'default'
  | 'info'
  | 'error'
  | 'success'
  | 'warning'
  | 'disabled';

export const stateToTextColor: Record<FieldState, string> = {
  info: 'text-info-1',
  error: 'text-error-1',
  success: 'text-success-1',
  warning: 'text-warning-1',
  default: 'text-grey-8',
  disabled: 'text-grey-6',
};

const stateToIcon: Record<FieldState, string> = {
  info: 'information-fill',
  error: 'spam-fill',
  success: 'checkbox-circle-fill',
  warning: 'alert-fill',
  default: '',
  disabled: '',
};

type Props = {
  /** Titre */
  title: string;
  /** Enfant: input, select... */
  children: React.ReactNode;
  /** Pour surcharger les styles du container */
  className?: string;
  /** Complément d'informations */
  hint?: string;
  /** Pour lier le libellé et le champ qu'il contient */
  htmlFor?: string;
  /** État */
  state?: FieldState;
  /** Message d'état affiché en dessous du champ */
  message?: string;
  /** Réduit la taille du titre */
  small?: boolean;
};

/** Wrapper pour élément de formulaire donnant des informations et un état */
export const Field = ({
  title,
  className,
  hint,
  htmlFor,
  state = 'default',
  message,
  children,
  small,
}: Props) => {
  return (
    <div className={classNames(`flex-grow flex flex-col gap-3`, className)}>
      <label htmlFor={htmlFor}>
        {/** Title */}
        <div
          className={classNames('font-medium text-grey-8', {
            'text-grey-5': state === 'disabled',
            'text-sm': small,
          })}
        >
          {title}
        </div>
        {/** Hint */}
        {hint !== undefined && (
          <span
            className={classNames('block mt-2 text-xs text-grey-7', {
              'text-grey-5': state === 'disabled',
            })}
          >
            {hint}
          </span>
        )}
      </label>
      {children}
      {message !== undefined && (
        <div
          className={classNames('flex items-center', stateToTextColor[state])}
        >
          {state !== 'disabled' && state !== 'default' && (
            <Icon icon={stateToIcon[state]} size="sm" className="mr-1" />
          )}
          <span className="text-xs">{message}</span>
        </div>
      )}
    </div>
  );
};
