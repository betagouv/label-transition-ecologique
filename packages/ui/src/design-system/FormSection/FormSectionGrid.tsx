import classNames from 'classnames';

export type FormSectionGridProps = {
  /** Enfants à afficher dans la grille (généralement des Field) */
  children: React.ReactNode;
  /** Permet d'ajuster les styles de la grille d'enfant */
  className?: string;
};

/**
 * Affiche une grille de champs de formulaire
 */
export const FormSectionGrid = ({
  children,
  className,
}: FormSectionGridProps) => (
  <div
    className={classNames('grid grid-cols-1 md:grid-cols-2 gap-6', className)}
  >
    {children}
  </div>
);
