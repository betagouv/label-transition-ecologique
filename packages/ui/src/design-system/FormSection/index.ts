export * from './FormSection';
export * from './FormSectionGrid';
