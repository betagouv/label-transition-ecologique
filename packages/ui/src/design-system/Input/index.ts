export * from './Input';
export * from './InputDate';
export * from './InputPassword';
