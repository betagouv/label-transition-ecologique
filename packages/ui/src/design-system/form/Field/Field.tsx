import classNames from 'classnames';
import {Icon} from '../../icons/Icon';

type State = 'default' | 'info' | 'error' | 'success' | 'warning' | 'disabled';

const stateToColor: Record<State, string> = {
  info: 'text-info-1',
  error: 'text-error-1',
  success: 'text-success-1',
  warning: 'text-warning-1',
  default: '',
  disabled: '',
};

const stateToIcon: Record<State, string> = {
  info: 'information-fill',
  error: 'spam-fill',
  success: 'checkbox-circle-fill',
  warning: 'alert-fill',
  default: '',
  disabled: '',
};

type Props = {
  /** Titre */
  title: string;
  /** Enfant: input, select... */
  children: React.ReactNode;
  /** Pour surcharger les styles du container */
  className?: string;
  /** Complément d'informations */
  hint?: string;
  /** Pour lier le libellé et le champ qu'il contient */
  htmlFor?: string;
  /** État */
  state?: State;
  /** Message d'état affiché en dessous du champ */
  message?: string;
};

/** Wrapper pour élément de formulaire donnant des informations et un état */
const Field = ({
  title,
  className,
  hint,
  htmlFor,
  state = 'default',
  message,
  children,
}: Props) => {
  return (
    <div className={classNames(`flex-grow flex flex-col gap-3`, className)}>
      <label htmlFor={htmlFor}>
        {/** Title */}
        <div
          className={classNames('font-medium text-grey-8', {
            '!text-grey-5': state === 'disabled',
          })}
        >
          {title}
        </div>
        {/** Hint */}
        {hint !== undefined && (
          <span
            className={classNames('block mt-2 text-xs text-grey-6', {
              '!text-grey-5': state === 'disabled',
            })}
          >
            {hint}
          </span>
        )}
      </label>
      {children}
      {message !== undefined && (
        <div className={classNames('flex items-center', stateToColor[state])}>
          <Icon icon={stateToIcon[state]} size="sm" className="mr-1" />
          <span className="text-xs">{message}</span>
        </div>
      )}
    </div>
  );
};

export default Field;
