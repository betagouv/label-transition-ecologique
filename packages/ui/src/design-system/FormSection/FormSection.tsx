import { Icon, IconValue } from '@tet/ui/design-system/Icon';
import { FormSectionGrid, FormSectionGridProps } from './FormSectionGrid';

type FormSectionProps = {
  /** Titre de la section */
  title: string;
  /** IcÃ´ne de la section */
  icon?: IconValue;
  /** Description optionnelle de la section */
  description?: string;
} & FormSectionGridProps;

/** Affiche une section de formulaire */
export const FormSection = ({
  className,
  title,
  description,
  icon,
  children,
}: FormSectionProps) => {
  return (
    <div className="flex flex-col gap-6">
      {/** Header */}
      <div className="flex flex-col gap-4">
        <div className="flex items-center gap-2">
          {icon && <Icon icon={icon} className="text-primary-7" />}
          <div className="font-bold text-primary-10">{title}</div>
        </div>
        {description && (
          <div className="text-sm text-primary-10">{description}</div>
        )}
        <div className="h-px bg-primary-3" />
      </div>
      {/** Grille de children */}
      <FormSectionGrid children={children} className={className} />
    </div>
  );
};
