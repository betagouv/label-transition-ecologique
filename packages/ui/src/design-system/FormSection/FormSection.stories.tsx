import {Meta, StoryObj} from '@storybook/react';

import {FormSection} from './FormSection';
import {Field} from '@design-system/Field';
import {Input} from '@design-system/Input';

const meta: Meta<typeof FormSection> = {
  component: FormSection,
  render: args => {
    return (
      <div className="w-full max-w-lg p-8">
        <FormSection title="Titre de la section" {...args}>
          {args.children}
        </FormSection>
      </div>
    );
  },
};

export default meta;

type Story = StoryObj<typeof FormSection>;

export const Default: Story = {};

export const AvecDescription: Story = {
  args: {
    description: 'Description de la section',
  },
};

export const AvecIcone: Story = {
  args: {
    description: 'Description de la section',
    icon: 'leaf-fill',
  },
};

export const AvecFields: Story = {
  args: {
    description: 'Description de la section',
    icon: 'leaf-fill',
    children: (
      <>
        <Field title="Titre du plan" className="md:col-span-2">
          <Input type="text" />
        </Field>
        <Field title="Statut">
          <Input type="date" />
        </Field>
        <Field title="Niveaude priorité">
          <Input type="date" />
        </Field>
      </>
    ),
  },
};

export const PlusieursSections: Story = {
  render: () => {
    return (
      <div className="w-full max-w-lg p-8 flex flex-col gap-10">
        <FormSection
          icon="leaf-fill"
          title="Titre de la section"
          description="Description de la section"
        >
          <Field title="Titre du plan" className="md:col-span-2">
            <Input type="text" />
          </Field>
          <Field title="Statut">
            <Input type="date" />
          </Field>
          <Field title="Niveaude priorité">
            <Input type="date" />
          </Field>
        </FormSection>
        <FormSection title="Le titre d'une autre section">
          <Field title="Titre du plan" className="md:col-span-2">
            <Input type="text" />
          </Field>
        </FormSection>
      </div>
    );
  },
};
