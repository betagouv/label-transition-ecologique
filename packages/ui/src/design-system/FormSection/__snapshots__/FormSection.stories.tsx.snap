// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design-system/FormSection AvecDescription smoke-test 1`] = `
<div class="w-full max-w-lg p-8">
  <div class="flex flex-col gap-6">
    <div class="flex flex-col gap-4">
      <div class="flex items-center gap-2">
        <div class="font-bold text-primary-10">
          Titre de la section
        </div>
      </div>
      <div class="text-sm text-primary-10">
        Description de la section
      </div>
      <div class="h-px bg-primary-3">
      </div>
    </div>
    <div class="grid grid-cols-1 md:grid-cols-2 gap-6">
    </div>
  </div>
</div>
`;

exports[`design-system/FormSection AvecFields smoke-test 1`] = `
<div class="w-full max-w-lg p-8">
  <div class="flex flex-col gap-6">
    <div class="flex flex-col gap-4">
      <div class="flex items-center gap-2">
        <span class="ri-leaf-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-primary-7">
        </span>
        <div class="font-bold text-primary-10">
          Titre de la section
        </div>
      </div>
      <div class="text-sm text-primary-10">
        Description de la section
      </div>
      <div class="h-px bg-primary-3">
      </div>
    </div>
    <div class="grid grid-cols-1 md:grid-cols-2 gap-6">
      <div class="flex-grow flex flex-col gap-3 md:col-span-2">
        <label>
          <div class="font-medium text-grey-8">
            Titre du plan
          </div>
        </label>
        <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
          <input type="text"
                 class="grow text-grey-8 px-4 outline-none text-md py-3"
          >
        </div>
      </div>
      <div class="flex-grow flex flex-col gap-3">
        <label>
          <div class="font-medium text-grey-8">
            Statut
          </div>
        </label>
        <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
          <input type="date"
                 class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4 [&amp;::-webkit-calendar-picker-indicator]:hidden"
          >
          <button type="button"
                  class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
          >
            <span class="ri-calendar-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
            </span>
          </button>
        </div>
      </div>
      <div class="flex-grow flex flex-col gap-3">
        <label>
          <div class="font-medium text-grey-8">
            Niveaude priorité
          </div>
        </label>
        <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
          <input type="date"
                 class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4 [&amp;::-webkit-calendar-picker-indicator]:hidden"
          >
          <button type="button"
                  class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
          >
            <span class="ri-calendar-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`design-system/FormSection AvecIcone smoke-test 1`] = `
<div class="w-full max-w-lg p-8">
  <div class="flex flex-col gap-6">
    <div class="flex flex-col gap-4">
      <div class="flex items-center gap-2">
        <span class="ri-leaf-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-primary-7">
        </span>
        <div class="font-bold text-primary-10">
          Titre de la section
        </div>
      </div>
      <div class="text-sm text-primary-10">
        Description de la section
      </div>
      <div class="h-px bg-primary-3">
      </div>
    </div>
    <div class="grid grid-cols-1 md:grid-cols-2 gap-6">
    </div>
  </div>
</div>
`;

exports[`design-system/FormSection Default smoke-test 1`] = `
<div class="w-full max-w-lg p-8">
  <div class="flex flex-col gap-6">
    <div class="flex flex-col gap-4">
      <div class="flex items-center gap-2">
        <div class="font-bold text-primary-10">
          Titre de la section
        </div>
      </div>
      <div class="h-px bg-primary-3">
      </div>
    </div>
    <div class="grid grid-cols-1 md:grid-cols-2 gap-6">
    </div>
  </div>
</div>
`;

exports[`design-system/FormSection PlusieursSections smoke-test 1`] = `
<div class="w-full max-w-lg p-8 flex flex-col gap-10">
  <div class="flex flex-col gap-6">
    <div class="flex flex-col gap-4">
      <div class="flex items-center gap-2">
        <span class="ri-leaf-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-primary-7">
        </span>
        <div class="font-bold text-primary-10">
          Titre de la section
        </div>
      </div>
      <div class="text-sm text-primary-10">
        Description de la section
      </div>
      <div class="h-px bg-primary-3">
      </div>
    </div>
    <div class="grid grid-cols-1 md:grid-cols-2 gap-6">
      <div class="flex-grow flex flex-col gap-3 md:col-span-2">
        <label>
          <div class="font-medium text-grey-8">
            Titre du plan
          </div>
        </label>
        <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
          <input type="text"
                 class="grow text-grey-8 px-4 outline-none text-md py-3"
          >
        </div>
      </div>
      <div class="flex-grow flex flex-col gap-3">
        <label>
          <div class="font-medium text-grey-8">
            Statut
          </div>
        </label>
        <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
          <input type="date"
                 class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4 [&amp;::-webkit-calendar-picker-indicator]:hidden"
          >
          <button type="button"
                  class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
          >
            <span class="ri-calendar-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
            </span>
          </button>
        </div>
      </div>
      <div class="flex-grow flex flex-col gap-3">
        <label>
          <div class="font-medium text-grey-8">
            Niveaude priorité
          </div>
        </label>
        <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
          <input type="date"
                 class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4 [&amp;::-webkit-calendar-picker-indicator]:hidden"
          >
          <button type="button"
                  class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
          >
            <span class="ri-calendar-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div class="flex flex-col gap-6">
    <div class="flex flex-col gap-4">
      <div class="flex items-center gap-2">
        <div class="font-bold text-primary-10">
          Le titre d'une autre section
        </div>
      </div>
      <div class="h-px bg-primary-3">
      </div>
    </div>
    <div class="grid grid-cols-1 md:grid-cols-2 gap-6">
      <div class="flex-grow flex flex-col gap-3 md:col-span-2">
        <label>
          <div class="font-medium text-grey-8">
            Titre du plan
          </div>
        </label>
        <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
          <input type="text"
                 class="grow text-grey-8 px-4 outline-none text-md py-3"
          >
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`design-system/FormSection SansHeader smoke-test 1`] = `
<div class="w-full max-w-lg p-8">
  <div class="grid grid-cols-1 md:grid-cols-2 gap-6">
    <div class="flex-grow flex flex-col gap-3 md:col-span-2">
      <label>
        <div class="font-medium text-grey-8">
          Titre du plan
        </div>
      </label>
      <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
        <input type="text"
               class="grow text-grey-8 px-4 outline-none text-md py-3"
        >
      </div>
    </div>
    <div class="flex-grow flex flex-col gap-3 md:col-span-2">
      <label>
        <div class="font-medium text-grey-8">
          Statut
        </div>
      </label>
      <div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
        <input type="date"
               class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4 [&amp;::-webkit-calendar-picker-indicator]:hidden"
        >
        <button type="button"
                class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
        >
          <span class="ri-calendar-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
